gnomad_qc/v4/annotations/insilico_predictors.py

```diff
@@ -159,6 +159,116 @@ def import_spliceai_vcf(path: str) -> hl.Table:
+        ).rows()
+        return ht
+
+    ht_g = import_pangolin_vcf(v4_genomes)
+    ht_e = import_pangolin_vcf(v4_exomes)
+
+    logger.info(
+        "Number of rows in genomes Pangolin HT: %s; "
+        "Number of rows in raw exomes Pangolin HT: %s. ",
+        ht_g.count(),
+        ht_e.count(),
+    )
+
+    ht = ht_g.union(ht_e)
+
+    logger.info("Exploding Pangolin scores...")
+    # `explode` will eliminate rows with empty array
+    # The VCF INFO of Pangolin is a string with the following format:
+    # gene1|pos_splice_gain:largest_increase|pos_splice_loss:largest_decrease|Warnings:||gene2...
+    # for example:
+    # Pangolin=ENSG00000121005.9|-86:0.25|38:-0.49|Warnings:||ENSG00000254238.1|-40:0.01|30:-0.17|Warnings:
+    # There's only one string element in the array before splitting, so we use
+    # [0] to extract the string, otherwise split can't be used on an array.
+    ht = ht.annotate(pangolin=ht.info.Pangolin[0].split(delim="\|\|"))
+    ht = ht.explode(ht.pangolin)
+    logger.info("Number of rows in exploded Pangolin Hail Table: %s", ht.count())
+
+    logger.info("Annotating Pangolin scores...")
+    # The Pangolin score is the delta score of splice gain and splice loss,
+    # which is the second and fourth element in the array after splitting.
+    ht = ht.transmute(
+        pangolin=hl.struct(
+            delta_scores=(
+                hl.empty_array(hl.tfloat64).append(
+                    hl.float(ht.pangolin.split(delim=":|\\|")[2])
+                )
+            ).append(hl.float(ht.pangolin.split(delim=":|\\|")[4])),
+        )
+    )
+
+    # Using > instead of >= in case where delta score of splice gain and splice
+    # loss are the same but different sign, we will keep the positive score of
+    # splice gain
+    ht = ht.annotate(
+        largest_ds_gene=hl.if_else(
+            hl.abs(hl.min(ht.pangolin.delta_scores))
+            > hl.abs(hl.max(ht.pangolin.delta_scores)),
+            hl.min(ht.pangolin.delta_scores),
+            hl.max(ht.pangolin.delta_scores),
+        )
+    )
+    ht = ht.select(ht.largest_ds_gene)
+    ht = ht.collect_by_key()
+    logger.info(
+        "Number of rows in Pangolin Hail Table after collect_by_key: %s", ht.count()
+    )
+    ht = ht.select(
+        pangolin=hl.struct(
+            largest_ds=hl.if_else(
+                hl.abs(hl.min(ht.values.largest_ds_gene))
+                > hl.abs(hl.max(ht.values.largest_ds_gene)),
+                hl.min(ht.values.largest_ds_gene),
+                hl.max(ht.values.largest_ds_gene),
+            )
+        )
+    )
+    logger.info(
+        "\nNumber of variants indicating splice gain: %s;\n"
+        "Number of variants indicating splice loss: %s; \n"
+        "Number of variants with no splicing consequence: %s \n",
+        ht.filter(ht.pangolin.largest_ds > 0).count(),
+        ht.filter(ht.pangolin.largest_ds < 0).count(),
+        ht.filter(ht.pangolin.largest_ds == 0).count(),
```

Comment:
```suggestion
        hl.agg.count_where(ht.pangolin.largest_ds > 0),
        hl.agg.count_where(ht.pangolin.largest_ds < 0),
        hl.agg.count_where(ht.pangolin.largest_ds == 0),
```
Yeah thats what I was thinking, I thought hail had an optimization on aggs over a filter(...).count(), like it may actually do all three of these at the same time but I'm not actually certain.